### PR TITLE
Update project skeleton and documentation

### PR DESCRIPTION
**Pull request by angelospanag**
A set of changes to bring this test up to date with the latest common practices in Python and Flask, there is no functional change to the core of the test and its requirements.

* Usage of `requirements.txt` instead of `setup.py` to install dependencies
* Usage of commonly ignored files/folders in `.gitignore` associated with Python, [as suggested by GitHub](https://github.com/github/gitignore/blob/master/Python.gitignore)
* Simplification of project skeleton, fixes on return types and usage of Flask 2.x features
* Replace jQuery functionality with vanilla JavaScript
* Updated documentation, fixed links and formatting

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
-* Python 3
+* [Python 3.7 or newer](https://www.python.org/)
 * Clone this repo
 * Access to the internet
 * Suitable development environment
 
 ## Getting Started
 
 * Create and activate a new virtual environment:
+    ```bash
```
Comment (interestinginteger123) [inline README.md:17]: This is sort of old hat these days as well

File: README.md [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
+    ```
 
 * Start the app:
-    $ python app.py
+    ```bash
+    flask run
+    ```
 
-* Visit http://localhost:5000 to receive the application.
+* Visit [`http://127.0.0.1:5000/`](http://127.0.0.1:5000/) to view the application.
```
Comment (interestinginteger123) [inline README.md:32]: Can still point to local? 127.0.0.1 is local host? Seems an unnecessary a change.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Flask==2.0.2
```
Comment (interestinginteger123) [inline requirements.txt:1]: Poetry lock instead of a requirements.txt I mean horses for courses

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+requests==2.27.1
+pytest==6.2.5
```
Comment (interestinginteger123) [inline requirements.txt:3]: There isn't a pytest in the readme originally as it was something that would be assessed in the test (tdd etc), assume it was left out originally as can use unit test instead etc (50/50 split in the community over pytest and unit test atm)

Comment (interestinginteger123) [reply requirements.txt:3]: Stand corrected was added after I joined. I mean if we want it super vanilla then it's all about the unit tests

File: README.md (hunk 2 above)
Comment (angelospanag) [reply README.md:32]: Just reused Flask's output but happy to change.

image

**Comment by angelospanag** [comment]
Closing this PR due to inactivity.

**Pull request closed without merging**